modules/llm-cache/ds/kv_cache_chunk.cc

```diff
@@ -0,0 +1,528 @@
+                                 RPCClient& rpc_client, int tensor_nbytes,
+                                 int layer, int chunk_size,
+                                 const std::string& kv_cache_ns,
+                                 ObjectID chunk_id) {
+  builder = std::make_shared<KVCacheChunkBuilder>(
+      rpc_client, tensor_nbytes, layer, chunk_size, kv_cache_ns);
+  builder->chunk_id_ = chunk_id;
+  return Status::OK();
+}
+
+Status KVCacheChunkBuilder::IsSame(const ObjectMeta& meta) {
+  RETURN_ON_ASSERT(is_ready_);
+  if (VLOG_IS_ON(100)) {
+    LOG(INFO) << "this: {total_tokens=" << total_tokens_
+              << ", chunk_size=" << chunk_size_
+              << ", tensor_nbytes=" << tensor_nbytes_ << ", layer=" << layer_
+              << ", md5=" << md5_ << "}";
+    LOG(INFO) << "that: " << meta.ToString();
+  }
+
+  RETURN_ON_ASSERT(meta.HasKey(KVCacheChunk::kFieldNameTotalTokens));
+  RETURN_ON_ASSERT(meta.HasKey(KVCacheChunk::kFieldNameChunkSize));
+  RETURN_ON_ASSERT(meta.HasKey(KVCacheChunk::kFieldNameTensorNBytes));
+  RETURN_ON_ASSERT(meta.HasKey(KVCacheChunk::kFieldNameLayer));
+  RETURN_ON_ASSERT(meta.HasKey(KVCacheChunk::kFieldNameAccessTime));
+  RETURN_ON_ASSERT(meta.HasKey(KVCacheChunk::kFieldNameMd5));
+  RETURN_ON_ASSERT(meta.HasKey(KVCacheChunk::kFieldNameNS));
+
+  RETURN_ON_ASSERT(meta.GetKeyValue<int>(KVCacheChunk::kFieldNameTotalTokens) ==
+                   total_tokens_);
+  RETURN_ON_ASSERT(meta.GetKeyValue<int>(KVCacheChunk::kFieldNameChunkSize) ==
+                   chunk_size_);
+  RETURN_ON_ASSERT(meta.GetKeyValue<int>(
+                       KVCacheChunk::kFieldNameTensorNBytes) == tensor_nbytes_);
+  RETURN_ON_ASSERT(meta.GetKeyValue<int>(KVCacheChunk::kFieldNameLayer) ==
+                   layer_);
+  // We assume it's not possilbe to have same name and md5 of all tokens
+  RETURN_ON_ASSERT(meta.GetKeyValue<std::string>(KVCacheChunk::kFieldNameMd5) ==
+                   md5_);
+
+  RETURN_ON_ASSERT(meta.GetKeyValue<std::string>(KVCacheChunk::kFieldNameNS) ==
+                   ns_);
+
+  return Status::OK();
+}
+
+Status KVCacheChunkBuilder::Query(
+    const std::vector<int>& prefix, const std::vector<int>& tokens,
+    std::vector<std::vector<std::pair<LLMKV, LLMKV>>>& kv_tensor) {
+  return QueryImpl(prefix, tokens, kv_tensor);
+}
+
+Status KVCacheChunkBuilder::Construct() {
+  ObjectMeta object_meta;
+  std::shared_ptr<Object> object = nullptr;
+  std::shared_ptr<KVCacheChunk> chunk = nullptr;
+
+  RETURN_ON_ASSERT(rpc_client_.Connected());
+
+  VLOG(100) << "Constructing " << ObjectIDToString(chunk_id_);
+  auto status = Status::OK();
+  status = rpc_client_.GetMetaData(chunk_id_, object_meta, true);
+  if (!status.ok()) {
+    VLOG(100) << "Get meta data failed: " << status.ToString();
+    return Status::ObjectNotExists();
+  } else if (object_meta.IsLocal()) {
+    object = rpc_client_.GetObject(chunk_id_);
+  }
+
+  // fetch from remote
+  if (object == nullptr) {
```

Comment:
This section (from line 140 to 147) could be better structured as:


```suggestion
  if (!status.ok()) {
    ...
    return Status::ObjectNotExists();
  }
  
  if (object_meta.IsLocal()) {
    object = rpc_client_.GetObject(chunk_id_);
    RETURN_ON_ASSERT(object != nullptr);
    return OkStatus();
  }
  
  // fetch from remote
  ...
```